lib/Runtime/Library/JavascriptRegExpConstructor.cpp

```diff
@@ -53,17 +54,44 @@ namespace Js
         this->lastInput = lastInput;
         this->lastMatch = lastMatch;
         this->reset = true;
+        this->invalidatedLastMatch = false;
+    }
+
+    void JavascriptRegExpConstructor::InvalidateLastMatch(UnifiedRegex::RegexPattern* lastPattern, JavascriptString* lastInput)
+    {
+        AssertMsg(lastPattern != nullptr, "lastPattern should not be null");
+        AssertMsg(lastInput != nullptr, "lastInput should not be null");
+        AssertMsg(JavascriptOperators::GetTypeId(lastInput) != TypeIds_Null, "lastInput should not be JavaScript null");
```

Comment:
Why these particular checks?

Reply:
We check similar things in `JavascriptRegExpConstructor::SetLastMatch` which is close to what we're doing here. Just kept it consistent.